File: frontend/src/types/medicine.ts
export interface MedicineItem {
  id: number;
  member_id: number;
  medicine_name: string;
  dosage: string | null;
  frequency: string | null;
  expiry_date: string | null;
  created_at: string;
  status: string;
}

export interface MedicineDetail {
  id: number;
  member_id: number;
  medicine_name: string;
  dosage: string | null;
  frequency: string | null;
  expiry_date: string | null;
  created_at: string;
  extracted_data: Record<string, string> | null;
}

export interface ExtractedMedicineData {
  medicine_name: string;
  extracted_fields: Record<string, string>;
}

export interface MedicineUploadResponse {
  message: string;
  medicine_id: number;
  extracted_data: ExtractedMedicineData;
}

export interface MedicineReminderItem {
  medicine_id: number;
  member_id: number;
  medicine_name: string;
  expiry_date: string | null;
  days_remaining: number;
}

File: app/schemas/medicine.py
from datetime import date
from pydantic import BaseModel


class MedicineCreate(BaseModel):
    member_id: int
    medicine_name: str
    dosage: str | None = None
    frequency: str | None = None
    expiry_date: date | None = None


class ExtractedMedicine(BaseModel):
    medicine_name: str
    extracted_fields: dict[str, str]